### ISSUE TITLE
crashes when zombie interacts with villager? bound must be positive | callDestructEntityEventDeath

### ISSUE THREAD

**Issue by johnfriedrich** [SpongePowered/Sponge#1298]
SF 2274

https://pastebin.com/F7VF2vdT

**Comment by waggz81** [comment]
I have been having the same issue with the last couple versions of Sponge/Forge. 

https://pastebin.com/CNBfvPYA

When I go to these locations I usually find a villager stuck in the ground, either legs or head. I end up killing them from the console now when I restart the server. They are never anywhere near any other players and shouldn't be in loaded chunks.

**Comment by gabizou** [comment]
So, I've tried to fix this issue with several different ways to go around it, at this point, I'm willing to put in some debugging information and just default these erroneous villagers to a default career while printing a message about their profession. My belief is that the Profession has no valid Career, in which the career itself cannot be resolved to populate a buying list, resulting in this exception.

That being said, I should have something out today for both 1.10.2 and 1.11.2 that will have some more information.

**Comment by waggz81** [comment]
Hi @gabizou , did you have an opportunity to get to this yet?

**Comment by sleiss** [comment]
I've added some debug to a custom version, it seems that some (all?) villagers of ExtraUtils2 have no career. These villagers are named "shady_merchants" and as they don't have a career, an error occurs while selecting one the careers (guess what, none is there..). 
I've added a workaround on a custom SpongeForge-Version (the relevant code is located in SpongeCommon) based on SF2300, however I'm not sure if I can just share the build with you, maybe a Sponge Contributor can help here.

Greets
